src/Workspaces/Core/Portable/Workspace/Host/Metadata/MetadataServiceFactory.cs

```diff
@@ -33,7 +38,7 @@ public MetadataFileReferenceProvider GetProvider()
 
             public PortableExecutableReference GetReference(string resolvedPath, MetadataReferenceProperties properties)
             {
-                return MetadataReference.CreateFromFile(resolvedPath, properties, _documentationService.GetDocumentationProvider(resolvedPath));
+                return (PortableExecutableReference)_metadataCache.GetReference(resolvedPath, properties);
```

Comment:
Cast seems odd here -- I'd expect the cache to always return PortableExecutableReferences, and MetadataReference.CreateFromFile already creates them. Do we need to update the type or make this generic or something?